src/test/scala/items/itemcategory/RepositoryTest.scala

```diff
@@ -0,0 +1,106 @@
+package io.github.pervasivecats
+package items.itemcategory
+
+import java.nio.file.attribute.UserPrincipalNotFoundException
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.FiniteDuration
+import scala.language.postfixOps
+
+import io.github.pervasivecats.items.ValidationError
+import io.github.pervasivecats.items.itemcategory.Repository.ItemCategoryNotFound
+import io.github.pervasivecats.items.itemcategory.Repository.OperationFailed
+import io.github.pervasivecats.items.itemcategory.entities.ItemCategory
+import io.github.pervasivecats.items.itemcategory.valueobjects.Description
+import io.github.pervasivecats.items.itemcategory.valueobjects.ItemCategoryId
+import io.github.pervasivecats.items.itemcategory.valueobjects.Name
+
+import com.dimafeng.testcontainers.JdbcDatabaseContainer.CommonParams
+import com.dimafeng.testcontainers.PostgreSQLContainer
+import com.dimafeng.testcontainers.scalatest.TestContainerForAll
+import org.scalatest.EitherValues.given
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers.*
+import org.testcontainers.utility.DockerImageName
+
+class RepositoryTest extends AnyFunSpec with TestContainerForAll {
+
+  private val timeout: FiniteDuration = 300.seconds
+
+  override val containerDef: PostgreSQLContainer.Def = PostgreSQLContainer.Def(
+    dockerImageName = DockerImageName.parse("postgres:15.1"),
+    databaseName = "items",
+    username = "test",
+    password = "test",
+    commonJdbcParams = CommonParams(timeout, timeout, Some("items.sql"))
+  )
+
+  @SuppressWarnings(Array("org.wartremover.warts.Var", "scalafix:DisableSyntax.var"))
+  private var repository: Option[Repository] = None
+
+  override def afterContainersStart(containers: Containers): Unit =
+    repository = Some(Repository.withPort(containers.container.getFirstMappedPort.intValue()))
```

Comment:
Unify this method with apply method creating a unique apply method taking a Config as a parameter